aws-amplify/amplify-cli
# Fix deployment state cleanup

The deployment-state.json is created during "iterative updates" in GraphQL schema deployments. This file is not deleted after completion of the deployment.  Future iterative updates fail if the deployment-state.json file exists but is not in "IDLE" state. 

#### Description of changes
The fix deletes the deployment-state.json file after the completion of any deployment ( success or failure ) . 




#### Issue #, if available
#9665 



#### Description of how you validated changes
manual tests:
1. Force iterative updates on a GraphQL schema.
2. Check deployment-state.json is getting created when deployment is in progress.
3. Validate that deployment-state.json is deleted after deployment concludes.

integration test:
schema-iterative-update-locking test succeeds

#### Checklist



- [*] PR description included
- [*] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
